### PR TITLE
Bug fixes and updates to documentation of designers

### PR DESCRIPTION
**Pull request by medinali**
Document all arguments of designers in the metadata

Change declare_population in simple_two_arm_designer

bug fixes to designers

add @author to all designers

### CODE REVIEW

File: R/regression_discontinuity_designer.R [hunk 1]
```diff
@@ -1,24 +1,17 @@
 #' Create a regression discontinuity design
-#' 
-#' Description: here.
-#' 
-#' Key limitations: here.
-#' 
-#' Note: here.
-#' 
+#'
```
Comment (jaspercooper) [inline R/regression_discontinuity_designer.R]: Fine to delete these placeholders for now but we can't forget to add these descriptions back into the documentation.

File: R/simple_two_arm_designer.R [hunk 2]
```diff
@@ -22,14 +30,16 @@ simple_two_arm_designer <- function(N = 100,
                                     treatment_sd = control_sd,
                                     rho = 1
 ){
-  
+  if(control_sd < 0 ) stop("control_sd must be non-negative")
+  if(prob < 0 | prob > 1) stop("prob must be in [0,1]")
+  if( rho < -1 | rho > 1) stop("rho must be in [-1,1]")
```
Comment (jaspercooper) [inline R/simple_two_arm_designer.R]: The proximity of the `<` and the `-` makes me nervous here. Can we do `abs(rho) > 1`?

File: vignettes/How_to_Write_and_Contribute_Designers.Rmd [hunk 3]
```diff
@@ -173,49 +173,40 @@ The documentation must be written as [Roxygen comments](http://r-pkgs.had.co.nz/
...
-#' @seealso \code{\link{spillover_design}} 
+#' @concept two arm design
+#' @seealso \code{\link{my_design}} \code{\link{simple_two_arm_designer}} 
 ```
 
 Finally, give yourself credit for your work.
 ```{r, eval=FALSE}
 #' 
-#' @author DeclareDesign Team
+#' @author  DeclareDesign Team \url{https://declaredesign.org/}
```
Comment (jaspercooper) [inline vignettes/How_to_Write_and_Contribute_Designers.Rmd]: I think we left out `@export` here

**Comment by jaspercooper** [comment]
Thanks @medinali this looks good -- can you take a look at the comments and push any changes you need to make.

File: tests/testthat/test_designers.R [hunk 4]
```diff
@@ -29,7 +29,7 @@ for(designer in designers){
   testthat::test_that(
     desc = paste0(designer,"'s default design runs."),
     code = {
-      diagnose_design(one_design,sims = 5,bootstrap = F)
+      expect_equal(class(diagnose_design(one_design,sims = 5,bootstrap = F)$diagnosands),"data.frame")
```
Comment (nfultz) [inline tests/testthat/test_designers.R]: `expect_is` I think?  https://stackoverflow.com/questions/48658381/what-is-the-difference-between-expect-is-and-expect-type-of-the-testthat-package

File: vignettes/How_to_Write_and_Contribute_Designers.Rmd (hunk 3 above)
Comment (jaspercooper) [reply vignettes/How_to_Write_and_Contribute_Designers.Rmd]: Great, thanks. @medinali would you mind adding some notes on roxygenizing prior to the PR?

**Pull request merged**